apps/prairielearn/src/lib/file-store.ts

```diff
@@ -120,41 +132,63 @@ export async function deleteFile(file_id, authn_user_id) {
 /**
  * Option of returning a stream instead of a file
  *
- * @param {number | string} file_id - The file to get.
- * @return {Promise<(import('stream'))>} - Requested file stream.
+ * @param file_id - The file to get.
+ * @return Requested file stream.
  */
-export async function getStream(file_id) {
+export async function getStream(file_id: number | string): Promise<ReadStream> {
   debug(`getStream(): file_id=${file_id}`);
   const file = await getFile(file_id, 'stream');
   return file.contents;
 }
 
+export async function getFile(
+  file_id: number | string,
+  data_type: 'stream',
+): Promise<{
+  contents: ReadStream;
+  file: File;
+}>;
+
+export async function getFile(
+  file_id: number | string,
+  data_type?: 'buffer',
+): Promise<{
+  contents: Buffer;
+  file: File;
+}>;
 /**
  * Get a file from the file store.
  *
- * @param {number | string} file_id - The file to get.
- * @return {Promise<object>} An object with a buffer (of the file contents) and a file object.
+ * @param file_id - The file to get.
+ * @return An object with a buffer (of the file contents) and a file object.
  */
-export async function getFile(file_id, data_type = 'buffer') {
+export async function getFile(
+  file_id: number | string,
+  data_type = 'buffer',
+): Promise<{
+  contents: Buffer | GetObjectOutput['Body'] | ReadStream;
+  file: File;
+}> {
   debug(`get(): file_id=${file_id}`);
   const params = { file_id };
-  const result = await sqldb.queryZeroOrOneRowAsync(sql.select_file, params);
+  const result = await sqldb.queryOptionalRow(sql.select_file, params, FileSchema);
   debug('get(): got row from DB');
 
-  let buffer, readStream;
+  let buffer: Buffer | GetObjectOutput['Body'];
+  let readStream: Buffer | GetObjectOutput['Body'] | ReadStream;
```

Comment:
These types don't look correct. Why could `readStream` be a Buffer? Why is `GetObjectOutput['Body']` involved here at all?

This could be resolved by my other suggestion to use early returns.